app/views/my_facilities/blood_pressure_control.html.erb

```diff
@@ -1,209 +1,318 @@
-              <div class="dropdown-divider"></div>
-              <h6 class="dropdown-header">
-                Monthly
-              </h6>
-              <% last_n_months(n: 6).each do |month| %>
-                <% selected = (month.year == registration_year && month.month == registration_month && cohort_period == :month) %>
-                <button
-                  class="dropdown-item query-filter-month <%= 'active' if selected %>"
-                  type="submit"
-                  data-month="<%= month.month %>"
-                  data-year="<%= month.year %>"
-                  form="query-filters"
-                >
-                  <%= month_short_name_and_year(month) %>
-                </button>
+        </div>
+        <% %w(large medium small community).each do |size| %>
+          <% next if @facilities_by_size[size].blank? %>
+          <p class="mt-20px mb-0px fw-bold">
+            <%= size.capitalize %> facilities
+          </p>
+          <table class="mt-3 mt-lg-4 table table-compact table-responsive-md table-hover analytics-table">
+            <colgroup>
+              <col>
+              <col>
+              <col>
+              <col class="table-divider">
+              <col>
+              <col class="table-divider">
+              <col>
+              <col class="table-divider">
+              <col>
+              <col class="table-divider">
+              <col>
+              <col class="table-divider">
+              <col>
+              <col class="table-divider">
+              <col>
+              <col class="table-divider">
+              <col>
+            </colgroup>
+            <thead>
+              <tr data-sort-method="thead" class="sorts">
+                <th class="row-label sort-label sort-label-small" style="width: 160px;">
+                  Facilities
+                </th>
+                <th class="row-label sort-label sort-label-small" data-sort-method="number" style="width: 120px;">
+                  Total assigned<br>patients
+                </th>
+                <th class="row-label sort-label sort-label-small" data-sort-method="number" style="width: 120px;">
+                  Total registered<br>patients
+                </th>
+                <th class="row-label sort-label sort-label-small" colspan="2" style="width: 80px;">
+                  6-month change
+                </th>
+                <% (@start_period..@period).each do |period| %>
+                  <th class="row-label sort-label sort-label-small" data-sort-method="number" colspan="2">
+                    <%= period %>
+                  </th>
+                <% end %>
+              </tr>
+            </thead>
+            <tbody>
+              <% @facilities_by_size[size].each do |facility| %>
+                <% six_month_rate_change = @data_for_facility[facility.name]["controlled_patients_rate"][@period] - @data_for_facility[facility.name]["controlled_patients_rate"][@start_period] || 0 %>
```

Comment:
This should be moved to a helper method.

Reply:
@rsanheim what's the value of moving the `a - b || 0` into it's own helper?

Reply:
Good question. One thing it does is keeps our views simple and more readable, which makes them easier to maintain in the future. We can also test helper functions, which probably isn't necessary for how simple this is but often these things get more complicated over time.

I haven't seen how you've changed things yet, but something like this would be a lot more readable:

```
  six_month_rate_change(facility)
```

or maybe

```
  six_month_rate_change(facility, "controlled_patients_rate")
```
if its needed for other rates.

Reply:
Thanks @rsanheim !